frame/dapps-staking/src/testing_utils.rs

```diff
@@ -575,17 +579,156 @@ pub(crate) fn assert_claim_staker(claimer: AccountId, contract_id: &MockSmartCon
     );
 }
 
+pub(crate) fn assert_claim_staker_with_delegate(claimer: AccountId, delegated_account: AccountId, contract_id: &MockSmartContract<AccountId>) {
+    let (claim_era, _) = DappsStaking::staker_info(&claimer, contract_id).claim();
+    let current_era = DappsStaking::current_era();
+
+    //clean up possible leftover events
+    System::reset_events();
+
+    let init_state_claim_era = MemorySnapshot::all(claim_era, contract_id, claimer);
+    let init_state_current_era = MemorySnapshot::all(current_era, contract_id, claimer);
+
+    let init_state_delegate_claim_era = MemorySnapshot::all(claim_era, contract_id, delegated_account);
```

Comment:
Would there be benefit of adding delegate to the memory snapshot struct?

Reply:
Yes, It would simplify the testing code. So not to make the delegating logic testing different from the other functions.